mne/time_frequency/tfr.py

```diff
@@ -245,6 +283,239 @@ def _cwt(X, Ws, mode="same", decim=1, use_fft=True):
         yield tfr
 
 
+# Loop of convolution: single trial
+
+
+def timefreq_transform(epoch_data, frequencies, sfreq=1.0, method='morlet',
+                       n_cycles=7.0, zero_mean=None, time_bandwidth=4.0,
+                       use_fft=True, decim=1, output='complex', n_jobs=1,
+                       verbose=None):
+    """Computes time frequency transforms.
+
+    Parameters
+    ----------
+    epoch_data : array of shape (n_epochs, n_channels, n_times)
+        The epochs.
+    frequencies : array-like of floats, shape (n_freqs)
+        The frequencies.
+    sfreq : float | int, defaults to 1.0
+        Sampling frequency of the data.
+    method : 'mtm' | 'morlet', defaults to 'morlet'
+        The time frequency method. 'morlet' convolves a Morlet wavelet. 'mtm'
+        convolves DPSS tapers.
+    n_cycles : float | array of float, defaults to 7.0
+        Number of cycles  in the Morlet wavelet. Fixed number
+        or one per frequency.
+    zero_mean : bool
+        Make sure the wavelets have a mean of zero. Defaults to True if
+        method == 'mtm' and defaults to False if method == 'morlet'.
+    time_bandwidth : float, defaults to 4.0 (3 tapers)
+        Time x (Full) Bandwidth product. Only applies if method == 'mtm'.
+        The number of good tapers (low-bias) is chosen automatically based on
+        this to equal floor(time_bandwidth - 1).
+    use_fft : bool, defaults to True
+        Use the FFT for convolutions or not.
+    decim : int | slice, defaults to 1
+        To reduce memory usage, decimation factor after time frequency
+        decomposition.
+        If `int`, returns tfr[..., ::decim].
+        If `slice` returns tfr[..., decim].
+        Note that decimation may create aliasing artifacts.
+    output : str, defaults to 'complex'
+        If 'complex', single trial complex.
+        If 'power', single trial power.
+        If 'phase', single trial phase.
+        If 'avg_power', average of single trial power.
+        If 'itc', inter-trial coherence.
+        If 'avg_power_itc', average of single trial power and inter-trial coherence across trials. # noqa
```

Comment:
I think this will render as a single line, can you check CircleCI?

I'd go with a bulleted list probably.